pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -337,6 +337,46 @@ func (configgen *ConfigGeneratorImpl) buildSidecarInboundListeners(env model.Env
 	return listeners
 }
 
+type listenerEntry struct {
+	service *model.Service
+	servicePort *model.Port
+	listener *xdsapi.Listener
+}
+
+func getServiceName(env model.Environment, service *model.Service) string {
+	attrs, err := env.GetServiceAttributes(service.Hostname)
+	if err != nil || attrs == nil || attrs.Name == "" {
+		return "unknown"
+	}
+	return fmt.Sprintf("%s/%s", attrs.Namespace, attrs.Name)
+}
```

Comment:
why do you need this? why isn't hostname sufficient?
This is also going to fail when you have a k8s service and a non k8s service hostname, to the extent that the error message you throw will be very confusing: "conflict with foo/bar and unknown". 

The service Hostname has all the elements needed to uniquely distinguish the service (foo.bar.svc.cluster.local). I don't think the extra name/namespace will add any additional information.

Reply:
I think this is also expensive ( env.GetServiceAttributes). 

Instead the namespace + name should be populated at conversion time, where they are available.